python/packagescanner.go

```diff
@@ -119,13 +119,37 @@ func (ps *Scanner) Scan(ctx context.Context, layer *claircore.Layer) ([]*clairco
 
 // FindDeliciousEgg finds eggs and wheels.
 func findDeliciousEgg(ctx context.Context, sys fs.FS) (out []string, err error) {
+	// Is this layer an rpm layer?
+	//
+	// If so, files in the disto-managed directory can be skipped.
+	var rpm bool
+	for _, p := range []string{
+		"var/lib/rpm/Packages",
+		"var/lib/rpm/rpmdb.sqlite",
+		"var/lib/rpm/Packages.db",
+	} {
+		if fi, err := fs.Stat(sys, p); err == nil && fi.Mode().IsRegular() {
+			rpm = true
+			break
+		}
+	}
+
 	return out, fs.WalkDir(sys, ".", func(p string, d fs.DirEntry, err error) error {
 		ev := zlog.Debug(ctx).
 			Str("file", p)
 		switch {
 		case err != nil:
 			ev.Discard().Send()
 			return err
+		case d.Name() == `dist-packages` && d.Type().IsDir():
```

Comment:
What happens in the situation when someone does:
```
# ubuntu:latest
apt-get update && apt-get install -y python3-pip && pip install urllib3
ls -l /usr/local/lib/python3.10/dist-packages/urllib3-1.26.14.dist-info/INSTALLER
```
Would this branch mean that we'd skip this package despite it not coming from an Ubuntu package?

Reply:
ah, yeah I misremembered how this works -- I'll fix it